### DISCUSSION TITLE
SQL err: Incorrect integer value when using UUID

### DISCUSSION THREAD

**Discussion by messerli90** [spatie/laravel-medialibrary#1606] [Q&A]
I could not associate images with a model that has `uuid` as the primary key using

`$community->addMediaFromRequest('media')->toMediaCollection('avatars');`

And getting the error:
`SQLSTATE[HY000]: General error: 1366 Incorrect integer value: 'f71279cd-6f33-4edc-a7fe-7f70cb9d039a' for column 'model_id' at row 1`

After some digging I found the migration uses:

`$table->morphs('model');`

I updated this to:

```
$table->string('model_id')->index();
$table->string('model_type')->index();
```

Everything seems to work fine now, but I'm not sure this is the *best* way to do it. Is there anything I should be aware of going forward doing it this way?

**Comment by LuCa-0** [comment]
I'm having the same issue and by changing the migration to a string (varchar) I get "/tmp/****" file does not exist or is not readable." error. Is there a way to fix?

**Comment by duyngha** [comment]
I got the same issue with `uuid`. Any updates?

**Comment by lunix15** [comment]
I found this <https://laracasts.com/discuss/channels/laravel/incorrect-integer-value-for-column-notifiable-id>
Maybe you should add this too:
`$table->index(['model_type', 'model_id']); // needed for uuid`

**Comment by dvigueras** [comment]
Replace `$table->morphs('model');`
with `$table->uuidMorphs('model');`
In your `create_media_table` migration and it will create a CHAR(36) column for the `model_id` field.
